plugins/inputs/prometheus/README.md

```diff
@@ -104,6 +100,35 @@ in Prometheus format.
 
 `urls` can contain a unix socket as well. If a different path is required (default is `/metrics` for both http[s] and unix) for a unix socket, add `path` as a query parameter as follows: `unix:///var/run/prometheus.sock?path=/custom/metrics`
 
+### Metric Format Configuration
+
+The `metric_version` setting controls how telegraf translates prometheus format
+metrics to telegraf metrics. There are two options.
+
+With `metric_version = 1`, the prometheus metric name becomes the telegraf
+metric name. Prometheus labels become telegraf tags. Prometheus values become
+telegraf field values. The fields have generic keys based on the type of the
+prometheus metric. This option produces metrics that are dense (not
+sparse). Denseness is a useful property for some outputs, including those that
+are more efficient with row-oriented data.
+
+`metric_version = 2` differs in a few ways. The prometheus metric name becomes a
+telegraf field key. Metrics hold more than one value and the field keys aren't
+generic. The resulting metrics are sparse, but for some outputs they may be
+easier to process or query, including those that are more efficient with
+column-oriented data. The telegraf metric name is the same for all metrics in
+the input instance. It can be set with the `name_override` setting and defaults
+to "prometheus". To have multiple metric names, you can use multiple instances
+of the plugin, each with its own `name_override`.
```

Comment:
Should we add a comment about round-trip-ability of this metric version versus version 1?

Reply:
I didn't write much on round tripping because I didn't have exact details on what works and doesn't work for both versions.  The serializer docs (https://github.com/influxdata/telegraf/blob/master/plugins/serializers/prometheus/README.md) imply that only 2 can round trip, but the input and output only say to make sure you use the same option in input and output if you want to round trip.

I was thinking it would be nice to add unit tests for round tripping if they don't already exist. That way it would be clear exactly what the shortcomings are and we could add clearer docs. I didn't want to hold up this PR on adding comprehensive unit tests though.

Reply:
Fair enough.